src/main/java/de/tum/in/www1/artemis/repository/ExamRepository.java

```diff
@@ -32,14 +32,30 @@ public interface ExamRepository extends JpaRepository<Exam, Long> {
     List<Exam> findByCourseId(long courseId);
 
     @Query("""
-            SELECT DISTINCT ex
+            SELECT ex
             FROM Exam ex
                 LEFT JOIN FETCH ex.exerciseGroups eg
                 LEFT JOIN FETCH eg.exercises
             WHERE ex.course.id = :courseId
             """)
     List<Exam> findByCourseIdWithExerciseGroupsAndExercises(@Param("courseId") long courseId);
 
+    // Find all exams for multiple courses (ids) that are already visible to the user (either registered, at least tutor or the exam is a test exam)
```

Comment:
Please add this as a JavaDoc comment instead.
Consider updating the other comments, too, if you have the time/nerves.